concepts/random/about.md

```diff
@@ -0,0 +1,128 @@
+
+The `Random` module provides two basic generators for generating integers and floats within a specific range:
+
+```elm
+generate 5 (Random.int -5 5)
+    --> [0, 3, -5, 5, 0]
+
+generate 3 (Random.float 0 5)
+    --> [1.61803, 3.14159, 2.71828]
+```
+
+Those values can be combined into pairs or lists of values:
+
+```elm
+generate 2 (Random.list 3 (Random.int 0 3))
+    --> [[0, 3, 3], [1, 3, 2]]
+
+generate 2 (Random.pair (Random.int 0 3) (Random.float 10 10.3))
+    --> [(0, 10.23412), (2, 10.17094)]
+```
+
+The (`elm-community/random-extra`)[random-extra] package provides a lot more generators for various data structures: strings, dates, dictionaries, arrays, sets, etc.
+
+We can create generators that will only return a single value using `Random.constant`:
+
+```elm
+generate 4 (Random.constant "hello")
+    --> ["hello", "hello", "hello", "hello"]
+```
+
+We can randomly pick from given elements with equal probability using `Random.uniform`:
+
+```elm
+generate 5 (Random.uniform Red [Green, Blue])
+    --> [Red, Blue, Blue, Green, Red]
+```
+
+`Random.uniform` takes two arguments to guarantee that there is at least one value to pick from.
+
+We can also tweak the probabilities using `Random.weighted`:
+
+```elm
+generate 5 (Random.weighted (Red, 80) [(Green, 15), (Blue, 5)])
+    --> [Red, Red, Green, Red, Red]
+```
+
+The values do not need to add up to 100, they will get renormalized anyway.
+
+We can reach the inside of a generator with `Random.map`:
+
+```elm
+generate 3 (Random.int 1 6 |> Random.map (\n -> n * 10))
+    --> [30, 60, 10]
+```
+
+We can also use `Random.map2` all the way to `Random.map5` to combine more generators:
+
+```elm
+position =
+  Random.map3
+    (\x y z -> Position x y z)
+    (Random.float -100 100)
+    (Random.float -100 100)
+    (Random.float -100 100)
+
+generate 1 position
+    --> [Position 33.788 -98.321 10.0845]
+```
+
+For more complex uses, we have `Random.andThen : (a -> Generator b) -> Generator a -> Generator b` that can use the value generated by one generator to create another:
+
+```elm
+bool = Random.uniform True [False]
+
+failHalfOfTheTime : Generator a -> Generator (Maybe a)
+failHalfOfTheTime generator =
+    bool
+        |> Random.andThen
+            (\boolResult ->
+                if boolResult then
+                    Random.map Just generator
+
+                else
+                    Random.constant Nothing
+            )
+
+generate 6 (Random.int 0 1 |> failHalfOfTheTime)
+    --> [Nothing, Just 1, Just 0, Nothing, Just 1, Nothing]
+```
+
+It is sometimes useful to define a generator self-recursively.
+In those cases, you might need to use `Random.lazy` to keep the compiler from unrolling the generator infinitely.
+
+```elm
+type Peano
+    = Zero
+    | Next Peano
+
+
+peano : Generator Peano
+peano =
+    Random.uniform (Random.constant Zero)
+        [ Random.lazy (\_ -> peano) |> Random.map Next
+        ]
+        |> Random.andThen identity
```

Comment:
Well, I did explain what `Random.andThen` does, and I think it's pretty clear what `identity` is.
But really, I don't want to explain this too much, because I feel it's the only "challenging" part which requires a bit of thought. And it's not really that challenging either since I show it here used in exactly the same way and also I left a hint.
I'll think about it some more :)

Reply:
I find this code pretty confusing as well, I can't say I understand at the moment. I do understand identity, and andThen, so maybe its lazy that I am finding confusing. From the comment showing the output, it looks like `generate 3` will create 3 uniformly distributed natural numbers between 0 and 2, is that right? And if so, what controls the range of the natural numbers produced?

Reply:
Maybe this example is too strange? I was trying to find one that's simple enough and different enough from the exercise one. Let me know if you have a better idea.

First of all, `lazy`doesn't add anything to the logic, all it does is wrap the function in a lambda so that the compiler doesn't infinitely stick `peano` into `peano`.

Then, what `peano` does is return `Zero` with a 50% chance , and itself plus one the other 50%. So the numbers coming out should be 0 (50% of the time), 1 (25% of the time), 2 (12% of the time), 3 (6% of the time), etc. The numbers are not bounded, because the generator could keep picking `peano` an arbitrary number of time, but it's more and more unlikely.

Maybe I should generate more numbers to get a better feel for the distribution.

Reply:
Ah I see, that does make sense, and this explanation is great :)

Maybe add it to the document?

Reply:
Done in [2b0732d](https://github.com/exercism/elm/pull/613/commits/2b0732de91c18562b7d9205a3f54202ff71ae382)